pathvalidate/_filepath.py

```diff
@@ -228,8 +251,7 @@ def validate(self, value: PathType) -> None:
         for entry in unicode_filepath.split("/"):
             if not entry or entry in (".", ".."):
                 continue
-
-            self.__fname_validator._validate_reserved_keywords(entry)
+            self.__fname_validator.validate(entry)
```

Comment:
This is mostly what we need to do to resolve the #51.
I think there was a reason why I chose `_validate_reserved_keywords()` instead of `validate()` here, but I can't remember what it was.

Adding `max_filename_len` to the filepath functions may be nice to have, but that seems like a different matter from #51.
Also, having `max_filename_len` and `max_filepath_len` args in the filename functions is confusing, so I'd like to avoid it.

Reply:
I think the problem will still exist if we just change this line, because the `__fname_validator` is created with `max_len=max_len`, which in the File**Path**Validator constructor refers to the max length of the filepath (so e.g. 4096 on Linux), while in the File**Name**Validator constructor refers to the max length of the filename (which should be less than 4096). Hence my reason for separating `max_len` into two newly named parameters. A solution without introducing new parameters would be to create `__fname_validator` with the reasonable default `max_len` of 255, however I would argue adding new parameters is a better solution as it allows more flexibility for the user. But, if you want to keep the `max_len` parameter, I can change to the first solution.

I agree having both `max_filename_len` and `max_filepath_len` is confusing for the filename functions. I can remove `max_filepath_len` for these as it is unnecessary.

Reply:
I guess, we can keep the `max_len` parameter, and just add a `max_filepath_len` for the filepath functions. It was just confusing to me that `max_len` means something different for filepath and filename methods. Let me know what you think the best way is.

Reply:
I think that it would be better to add `max_filename_len` and `max_filepath_len` arguments to the filepath functions and leave `max_len` argument as an alias for `max_filepath_len` rather than deprecating it.